lib/hardware/modules/iob_demux/iob_demux.py

```diff
@@ -3,7 +3,74 @@ def setup(py_params_dict):
         "original_name": "iob_demux",
         "name": "iob_demux",
         "version": "0.1",
-        "generate_hw": False,
+        "confs": [
+            {
+                "name": "DATA_W",
+                "type": "P",
+                "val": "21",
+                "min": "0",
+                "max": "NA",
+                "descr": "",
+            },
+            {
+                "name": "N",
+                "type": "P",
+                "val": "21",
+                "min": "0",
+                "max": "NA",
+                "descr": "",
+            },
+        ],
+        "ports": [
+            {
+                "name": "sel",
+                "descr": "Input port",
+                "signals": [
+                    {
+                        "name": "sel",
+                        "width": "($clog2(N)+($clog2(N)==0))",
+                        "direction": "input",
+                    },
+                ],
+            },
+            {
+                "name": "data_i",
+                "descr": "Input port",
+                "signals": [
+                    {
+                        "name": "data",
+                        "width": "N*DATA_W",
+                        "direction": "input",
+                    },
+                ],
+            },
+            {
+                "name": "data_o",
+                "descr": "Output port",
+                "signals": [
+                    {
+                        "name": "data",
+                        "width": "N*DATA_W",
+                        "direction": "output",
+                    },
+                ],
+            },
+        ],
+        "snippets": [
+            {
+                "outputs": ["data_o"],
```

Comment:
The 'outputs' list of the snippets were originally intended to reference signals (like the signal `data`) and not groups of signals (like the port `data_o`).
I don't think we can reference entire ports, because some of them may contain input signals as well.

However, trying to reference "data" here would become ambiguous, since we have two signals with the name "data" (in different groups/ports).

Reply:
I'm not sure how to solve this.
This issue is related to: https://github.com/IObundle/iob-soc/pull/865#discussion_r1683650502

Reply:
can we reference wire/port.signal ?

Reply:
In a previous commit we were planning to be able to specify the wire/port from which the referenced signal came from, using the "wire" attribute:
https://github.com/IObundle/iob-soc/blob/c99f767cb8cc0839a8e96a266384c8a6796697e7/lib/hardware/modules/iob_pulse_gen/iob_pulse_gen.py#L114-L115
In the above example, we are referencing the signal "start_detected_nxt" from the wire "start_detected_nxt".

We deprecated the option to specify the wire in this commit:
https://github.com/IObundle/iob-soc/commit/30ece37874567fb7de350e94905266f55497940a#diff-e6adb0ff5290fcf15b7fbf94f7f370e3cef12f0210079593f3664165092ca2eaL114-R113

We could revert that commit to be able to specify which wire/port we are referencing.

Reply:
> can we reference wire/port.signal ?

We could do that as well, by updating the py2hwsw scripts to support that syntax `<wire/port name>.<signal name>`.
The `iob_demux.py` snippet we become:
```suggestion
                "outputs": ["data_o.data"],
```
Port name = "data_o"
Signal name = "data"

Reply:
port name does not have suffix

Reply:
In the current PR state it does have a suffix. See [here](https://github.com/IObundle/iob-soc/pull/866/files/ad43988ac54cf2f27d83ba6e6e025c668afbe91f#diff-039f692536a7b07fa298628979e24db9b582be4aae8ed2f4a74befca60070254R48).
Maybe that suffix should be removed.